state-synchronizer/src/logging.rs

```diff
@@ -0,0 +1,153 @@
+use libra_logger::Schema;
+use libra_types::{
+    contract_event::ContractEvent, ledger_info::LedgerInfoWithSignatures, waypoint::Waypoint,
+};
+use serde::Serialize;
+#[derive(Clone, Schema)]
+pub struct LogSchema<'a> {
+    name: LogEntry,
+    event: Option<LogEvent>,
+    #[schema(debug)]
+    error: Option<&'a Error>,
+    #[schema(display)]
+    peer: Option<PeerNetworkId>,
+    is_upstream_peer: Option<bool>,
+    #[schema(display)]
+    chunk_request: Option<GetChunkRequest>,
+    version: Option<u64>,
+    #[schema(display)]
+    chunk_response: Option<GetChunkResponse>,
+    #[schema(display)]
+    waypoint: Option<Waypoint>,
+    subscription_name: Option<String>,
+    count: Option<usize>,
+    reconfig_events: Option<Vec<ContractEvent>>,
+    local_li_version: Option<u64>,
+    local_synced_version: Option<u64>,
+    local_epoch: Option<u64>,
+    #[schema(display)]
+    ledger_info: Option<LedgerInfoWithSignatures>,
+    old_epoch: Option<u64>,
+    new_epoch: Option<u64>,
+    target_version: Option<u64>,
+}
+
+impl<'a> LogSchema<'a> {
+    pub fn new(name: LogEntry) -> Self {
+        Self {
+            name,
+            event: None,
+            peer: None,
+            is_upstream_peer: None,
+            error: None,
+            chunk_request: None,
+            chunk_response: None,
+            version: None,
+            waypoint: None,
+            subscription_name: None,
+            reconfig_events: None,
+            count: None,
+            local_li_version: None,
+            local_synced_version: None,
+            local_epoch: None,
+            ledger_info: None,
+            old_epoch: None,
+            new_epoch: None,
+            target_version: None,
+        }
+    }
+
+    pub fn event_log(name: LogEntry, event: LogEvent) -> Self {
+        Self {
+            name,
+            event: Some(event),
+            peer: None,
+            is_upstream_peer: None,
+            error: None,
+            chunk_request: None,
+            chunk_response: None,
+            version: None,
+            waypoint: None,
+            subscription_name: None,
+            reconfig_events: None,
+            count: None,
+            local_li_version: None,
+            local_synced_version: None,
+            local_epoch: None,
+            ledger_info: None,
+            new_epoch: None,
+            old_epoch: None,
+            target_version: None,
+        }
+    }
+
+    pub fn chunk_req(mut self, request: &GetChunkRequest) -> Self {
+        self.chunk_request = Some(request.clone());
+        self
+    }
+
+    pub fn chunk_resp(mut self, response: &GetChunkResponse) -> Self {
+        self.chunk_response = Some(response.clone());
+        self
+    }
+
+    pub fn peer_id(mut self, peer_id: &PeerNetworkId) -> Self {
+        self.peer = Some(peer_id.clone());
+        self
+    }
+}
+
+#[derive(Clone, Copy, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum LogEntry {
```

Comment:
do you need this? there's already those metadata in logs 
```
            module_path!().split("::").next().unwrap(),
            module_path!(),
            file!(),
            line!(),
            concat!(file!(), ':', line!()),
```

Reply:
my reasoning was that the LogEntry (and LogEvent) would help search for specific logs when we debug, can sort of serve like IDs for logs

Reply:
it is a mixed bag, the advantage of having this well defined is that even if you move around the log lines or rename the functions these fields are static and very descriptive of the intent!